src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagDisabledITest.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.companieshouse.pscdataapi.api;
+
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.*;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.chskafka.ChangedResource;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.chskafka.PrivateChangedResourceHandler;
+import uk.gov.companieshouse.api.handler.chskafka.request.PrivateChangedResourcePost;
+import uk.gov.companieshouse.api.http.HttpClient;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.sdk.ApiClientService;
+import uk.gov.companieshouse.pscdataapi.util.TestHelper;
+
+@SpringBootTest
+class ResourceChangedApiServiceAspectFeatureFlagDisabledITest {
+
+    @InjectMocks
+    private ChsKafkaApiService chsKafkaService;
+
+    @MockBean
+    private ApiClientService apiClientService;
+
+    @Mock
+    private InternalApiClient internalApiClient;
+
+    @Mock
+    private ChangedResource changedResource;
+    @Mock
+    private PrivateChangedResourceHandler privateChangedResourceHandler;
+    @Mock
+    private PrivateChangedResourcePost changedResourcePost;
+    @Mock
+    private ApiResponse<Void> response;
+    @Mock
+    private HttpClient httpClient;
+
+    @MockBean
+    private ChsKafkaApiService mapper;
+
+    private TestHelper testHelper;
+
+    @Captor
+    ArgumentCaptor<ChangedResource> changedResourceCaptor;
+
+
+
+    @BeforeEach
+    void setup() {
+        when(internalApiClient.getHttpClient()).thenReturn(httpClient);
+        testHelper = new TestHelper();
+    }
+
+    @Test
+    void testThatKafkaApiShouldBeCalledWhenFeatureFlagDisabled()
```

Comment:
I tried making the test fail by setting the feature flag to true in **feature.seeding_collection_enabled** in **application.properties**. I also tried adding **@ActiveProfiles("feature_flag_enabled")** to the class, similar to the other test class. However, the test still passed, which means we are not properly testing the feature flag

Reply:
I'm happy that the testThatAspectShouldNotProceedWhenFeatureFlagEnabled test cover this test case

Reply:
The @InjectMocks annotation bypasses the Aspect which means the Disabled test cannot fail, but the InjectMocks is required due to the structure of the API. The Enabled test does pass/fail when we want it to. Confirmation from James Andrew that we're happy to proceed with this.